backend/src/main/java/neuefische/capstone/bpmndiagram/BpmnDiagramController.java

```diff
@@ -40,13 +40,14 @@ public BpmnDiagram addBpmnDiagram(@RequestBody @Valid BpmnDiagram newBpmnDiagram
 
     @PutMapping("/{id}")
     public BpmnDiagram updateBpmnDiagram(@RequestBody @Valid BpmnDiagram updatedBpmnDiagram, @PathVariable String id) {
-        if(updatedBpmnDiagram.id().equals(id)) {
-            try {
-                return service.updateBpmnDiagram(updatedBpmnDiagram);
-            } catch (NoSuchElementException e) {
-                throw new ResponseStatusException(HttpStatus.NOT_FOUND, e.getMessage());
-            }
-        } else throw new ResponseStatusException(HttpStatus.BAD_REQUEST,"Request ID Mismatch");
+        if (!updatedBpmnDiagram.id().equals(id)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Request ID Mismatch");
```

Comment:
Instead of passing both id and BpmnDiagram you could just pass your BpmnDiagram to update from your frontend. Then you can omit the check for matching ids

Reply:
Yes, but i think it's more REST-conform with ID PathVariable...